lib/stdlib/src/erl_lint.erl

```diff
@@ -2505,14 +2509,23 @@ gexpr({op,_Anno,EqOp,L,R}, Vt, St0) when EqOp =:= '=:='; EqOp =:= '=/=' ->
 gexpr({op,Anno,Op,L,R}, Vt, St0) ->
     {Avt,St1} = gexpr_list([L,R], Vt, St0),
     case is_gexpr_op(Op, 2) of
-        true -> {Avt,St1};
+        true -> {Avt,warn_obsolete_op(Op, 2, Anno, St1)};
         false -> {Avt,add_error(Anno, illegal_guard_expr, St1)}
     end;
 %% Everything else is illegal! You could put explicit tests here to
 %% better error diagnostics.
 gexpr(E, _Vt, St) ->
     {[],add_error(element(2, E), illegal_guard_expr, St)}.
 
+warn_obsolete_op(Op, A, Anno, St) ->
+    case {Op, A} of
+        {'and', 2} ->
+            add_warning(Anno, {deprecated, "'and'", "use 'andalso' instead", "OTP 29"}, St);
+        {'or', 2} ->
+            add_warning(Anno, {deprecated, "'or'", "use 'orelse' instead", "OTP 29"}, St);
+        _ -> St
+    end.
```

Comment:
Please add a way to disable this warning, and keep it disabled by default for now. Regardless of what decision we're going to make, I still see it as a useful optional warning.

Reply:
@richcarl If you do this, and rebase the branch, we can include this in OTP 29. (Feel free to enable the warning for all code in OTP.)